### PR TITLE
Document flatten

### PR DESCRIPTION
**Pull request by johlrogge**

### CODE REVIEW

File: src/flatten.rs [hunk 1]
```diff
@@ -6,9 +6,54 @@
 
 /// Extension trait for `Iterator<Item = Result<O, E>>` to selectively transform Oks and Errors.
 pub trait Flatten<O, E>: Sized {
+    /// [flatten](Iterator::flatten) `Err` values while leaving `Ok`-values as is.
```
Comment (matthiasbeyer) [inline src/flatten.rs]: Isn't it the other way round?

File: src/flatten.rs [hunk 2]
```diff
@@ -6,9 +6,54 @@
...
+    /// assert_eq!(
+    ///     mapped,
+    ///     [Ok(0), Ok(0), Ok(1), Err(0..4), Err(0..0), Ok(0), Ok(1)]
+    /// );
+    /// ```
     fn flatten_ok<U, O2>(self) -> FlattenOk<Self, U>
     where
         U: IntoIterator<Item = O2>;
+
+    /// [flatten](Iterator::flatten) `Ok` values while leaving `Err`-values as is.
```
Comment (matthiasbeyer) [inline src/flatten.rs]: Same here...?

File: src/flatten.rs (hunk 1 above)
Comment (johlrogge) [reply src/flatten.rs]: Of course, I must have been tired trying to get everything done yesterday. Will fix,

File: src/flatten.rs (hunk 2 above)
Comment (johlrogge) [reply src/flatten.rs]: indeed

**Review by matthiasbeyer — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
bors merge

**Pull request merged**